volume/store/store.go

```diff
@@ -47,39 +60,94 @@ func (s *VolumeStore) AddAll(vols []volume.Volume) {
 func (s *VolumeStore) Create(name, driverName string, opts map[string]string) (volume.Volume, error) {
 	s.mu.Lock()
 	if vc, exists := s.vols[name]; exists {
-		v := vc.Volume
 		s.mu.Unlock()
-		return v, nil
+		<-vc.waitCreate
+		if vc.Volume != nil && (vc.Volume.DriverName() == driverName || driverName == volume.DefaultDriverName) {
```

Comment:
@cpuguy83  Don't understand the logic `driverName == volume.DefaultDriverName`. Does it mean if the 
`vc.Volume.DriverName()` is `convoy`, but the `driverName` is `local` and then return the `volume`?

And one question, from what I understand `VolumeStore` only store the volumes which has been reference by
containers, so there is the situation, a volume exists in external volume driver but not stored in `VolumeStore`, so the `Create` will create a volume has the same name with these volumes. So we can't ensure the volume has a unique name. I think it's better to check if the volume is exist in external volume driver, but anyway, this may take long time to create a volume.  Am I missing something?